cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/connector/ExploreDetail.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.etl.api.connector;
+
+import io.cdap.cdap.api.plugin.PluginPropertyField;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * The explore result for the given request
+ */
+public class ExploreDetail {
+  // this count represents the total count of entities, when pagination is added in the future,
+  // this count might not be equal to entities.size()
+  private final int count;
+  // explore entity type -> sample property
+  private final Map<String, Set<PluginPropertyField>> sampleProperties;
```

Comment:
it's normally better to avoid Maps in these objects, as they are less extensible. A collection of objects is preferred, even if that object just has 2 fields for now (entity type and set of properties)

Reply:
PluginPropertyField is normally not something we want a plugin to construct, but something that gets generated automatically by examining the PluginConfig class. Creating directly through the constructor is pretty error prone.

Is there a way for the plugin to get this somehow or would it need to create it by calling the constructor?